js/src/plugins/server-event.plugin.js

```diff
@@ -8,33 +8,44 @@ export default class serverEvent extends atkPlugin {
 
     const element = this.$el;
     const hasLoader = this.settings.showLoader;
+    const that = this;
 
     if (typeof(EventSource) !== "undefined") {
-      let source = new EventSource(`${this.settings.uri}&__atk_sse=1`);
+      this.source = new EventSource(`${this.settings.uri}&__atk_sse=1`);
       if(hasLoader) {
         element.addClass('loading');
       }
-      source.onmessage = function (e) {
+      this.source.onmessage = function (e) {
         apiService.atkSuccessTest(JSON.parse(e.data));
       };
-      source.onerror = function (e) {
+      this.source.onerror = function (e) {
           if (e.eventPhase === EventSource.CLOSED) {
             if (hasLoader) {
               element.removeClass('loading');
             }
-            source.close();
+            that.source.close();
           }
       };
-      source.addEventListener("jsAction", function(e) {
+      this.source.addEventListener("jsAction", function(e) {
         apiService.atkSuccessTest(JSON.parse(e.data));
       }, false);
+      window.addEventListener('beforeunload', function(event) {
+        that.source.close();
```

Comment:
@abbadon1334 What is the use case when this is needed?

Reply:
in a long run SSE, if you click the browser reload button, the browser will not reload, there is a sort of locking on browser side for the connection.
if you can, try the above code to reproduce the issue.

Reply:
Ok, interesting, as F5 should always work... It does not work probably as the browser think that the original page is still not loaded yet due SSE.

Reply:
> Ok, interesting, as F5 should always work... It does not work probably as the browser think that the original page is still not loaded yet due SSE.

yes if you search there are many stackoverflow post about it.

I think that 2 php lines, can be added directly in class jsSSE to check if connection is still active or not after every flush of a push data.

This way at least on the next push of data the browser will be unlocked.

Reply:
@abbadon1334 - I have made some changes and check for an aborted connection in jsSSE class.
Also, I add an option for the plugin to close connection on beforeUnload event. It is off by default but it can be turned on easily. 

Beside, I was trying to reproduce your issue without success, so not sure if I was doing it properly.
I did not have problem reloading the browser window at all.

Let me know what you think.

Reply:
@ibelar , @abbadon1334 Do you use the same browser?

I think reload should never be blocked, at least without any warning. So consider always disconnectiong (not aborting) the SSE.

Reply:
This pr should currently allow for customization. @abbadon1334 - let us know if you need more changes.